### PR TITLE
Change order of debugging tools to show recommended first.

### PR DESCRIPTION
**Pull request by polina-c**
People request better documentation for debugging in DevTools, thinking it is first recommended tool.

### CODE REVIEW

File: src/testing/debugging.md [hunk 1]
```diff
@@ -25,6 +23,9 @@ Flutter applications. Here are some of the available tools:
 * If you are looking for a way to use GDB to remotely debug the
   Flutter engine running within an Android app process,
   check out [`flutter_gdb`][].
+* [DevTools][], a suite of performance and profiling
+  tools that run in a browser.
+
```
Comment (kenzieschmoll) [inline src/testing/debugging.md:28]: I would move this before the Flutter Inspector bullet, since the inspector section assumes awareness of DevTools

**Review by kenzieschmoll — APPROVED** [review]
LGTM with one comment

**Review by sfshaza2 — APPROVED** [review]
lgtm, once Kenzie is happy

File: src/testing/debugging.md [hunk 2]
```diff
@@ -8,275 +8,26 @@ description: How to debug your app using the DevTools suite.
...
-[release mode][], as the debugging and profiling
-information has been stripped away.
-
-If you use DevTools for profiling, make sure to
-run your application in [profile mode][]. Otherwise,
-the main output that appears on your profile are the
-debug asserts verifying the framework's various invariants
-(see [Debug mode assertions](#debug-mode-assertions)).
-
-![GIF showing DevTools features]({{site.url}}/assets/images/docs/tools/devtools/inspector.gif){:width="100%"}
```
Comment (kenzieschmoll) [inline src/testing/debugging.md:60]: this gif can be deleted too

File: src/testing/debugging.md [hunk 3]
```diff
@@ -8,275 +8,26 @@ description: How to debug your app using the DevTools suite.
...
-  tools that run in a browser.
-* [Android Studio/IntelliJ][], and [VS Code][]
+* [VS Code][](recommended) and [Android Studio/IntelliJ][], 
   (enabled with the Flutter and Dart plugins)
   support a built-in source-level debugger with
   the ability to set breakpoints, step through code,
   and examine values.
+* If you are looking for a way to use GDB to remotely debug the
+  Flutter engine running within an Android app process,
+  check out [`flutter_gdb`][].
```
Comment (kenzieschmoll) [inline src/testing/debugging.md:18]: I think this should still be last as this is the lest relevant for users

File: src/testing/debugging.md [hunk 4]
```diff
@@ -8,275 +8,26 @@ description: How to debug your app using the DevTools suite.
...
-test your code with `flutter analyze`.
-
-The Dart analyzer makes heavy use of type annotations that
-you put in your code to help track problems down.
-You are encouraged to use them everywhere (avoiding `var`,
-untyped arguments, untyped list literals, and so on)
-as this is the quickest and least painful way of tracking
-down problems.
-
-For more information, see [Using the Dart analyzer][].
```
Comment (kenzieschmoll) [inline src/testing/debugging.md:88]: I think this is still relevant

File: src/testing/debugging.md [hunk 5]
```diff
@@ -8,275 +8,26 @@ description: How to debug your app using the DevTools suite.
...
-This allows developers to enable or disable invariant
-checking, such that the associated performance cost
-is only paid during debugging sessions.
-
-When an invariant is violated, it's reported to the
-console, with some context information to help track
-down the source of the problem.
-
-For more information, check out [Assert][] in the
-[Dart language documentation][].
```
Comment (kenzieschmoll) [inline src/testing/debugging.md:132]: still relevant?

File: src/testing/debugging.md [hunk 6]
```diff
@@ -8,275 +8,26 @@ description: How to debug your app using the DevTools suite.
...
-For example:
-
-```
-{
-  "engineEnterTimestampMicros": 96025565262,
-  "timeToFirstFrameMicros": 2171978,
-  "timeToFrameworkInitMicros": 514585,
-  "timeAfterFrameworkInitMicros": 1657393
-}
-```
```
Comment (kenzieschmoll) [inline src/testing/debugging.md:172]: still relevant? @sfshaza2  is all this deleted documentation tracked somewhere else on the flutter website?

File: src/testing/debugging.md [hunk 7]
```diff
@@ -8,275 +8,26 @@ description: How to debug your app using the DevTools suite.
...
-## Debug flags
-
-In most cases, you won't need to use the debug flags
-directly, as you'll find the most useful debugging
-functionality in the [DevTools][] suite. But if you
-prefer to use the debug flags directly, see
-[Debug flags: performance][] in the
-[Debugging Flutter apps programmatically][] page.
-
-## Common problems
```
Comment (kenzieschmoll) [inline src/testing/debugging.md:213]: is this whole section still relevant?

File: src/testing/debugging.md (hunk 4 above)
Comment (polina-c) [reply src/testing/debugging.md:88]: Yes, this is relevant. But it should be incorporated into new content / structure. 
I commented on the issue to take some deleted content from this PR.

File: src/testing/debugging.md (hunk 5 above)
Comment (polina-c) [reply src/testing/debugging.md:132]: see previous comment

File: src/testing/debugging.md (hunk 6 above)
Comment (polina-c) [reply src/testing/debugging.md]: Linked this PR to the issue: see previous comment

File: src/testing/debugging.md (hunk 7 above)
Comment (polina-c) [reply src/testing/debugging.md:213]: see previous comment

File: src/testing/debugging.md [hunk 8]
```diff
@@ -287,10 +38,6 @@ You might find the following docs useful:
 * [Use a native debugger][]
 * [Flutter's modes][]
 * [Debugging Flutter apps programmatically][]
-* [DevTools][]
-* [Android Studio/IntelliJ][]
-* [VS Code][]
-
 
 [Flutter enabled IDE/editor]: {{site.url}}/get-started/editor
```
Comment (kenzieschmoll) [inline src/testing/debugging.md:42]: please delete site links in this bottom section that are no longer used

**Review by kenzieschmoll — APPROVED** [review]
one comment then lgtm

**Pull request merged**